tuf/client/client.go

```diff
@@ -505,9 +518,9 @@ func (c Client) RoleTargetsPath(role string, hashSha256 string, consistent bool)
 	return role, nil
 }
 
-// TargetMeta ensures the repo is up to date, downloading the minimum
-// necessary metadata files
-func (c Client) TargetMeta(path string) (*data.FileMeta, error) {
+// TargetMeta ensures the repo is up to date. It assumes downloadTargets
+// has already downloaded all delegated roles
+func (c Client) TargetMeta(role, path string) (*data.FileMeta, error) {
 	c.Update()
```

Comment:
I know you didn't change this line, but I think the only thing that calls this might be `GetTargetName`, which does a `c.Update` right before this.  Is it necessary to update again before getting the metadata for every role?  If so, should we check for an error at least and return it?